src/guides/protocol.md

```diff
@@ -0,0 +1,230 @@
+
+  - merkle: Base transaction data
+
+  - witness: Transaction witness data
+
+  - tree: Root of the name-resource urkel tree. (updated every
+  [36 blocks, about 4 times a day](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/protocol/networks.js#L295))
+
+  - filter: (Not currently implemented) Will be used to commit to
+  [Neutrino-style filters](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki)
+
+  - reserved: (Not currently implemented) Will be used for layer 2 expansion.
+
+- Coinbase transactions:
+
+    - Because Handshake is 100% SegWit, there are no `ScriptSig`'s, just witness data.
+    Therefore,
+    [to enforce transaction uniqueness](https://github.com/bitcoin/bips/blob/master/bip-0034.mediawiki),
+    block heights are committed to the
+    [`locktime` value of coinbase transactions.](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/mining/template.js#L254-L255)
+
+    - [Airdrops](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/mining/template.js#L310-L324)
+    and [Claims](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/mining/template.js#L282-L308)
+    are included in the coinbase transaction. Proofs for both of these types are
+    [verified during block contextual checks, in `tx.js`.](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/primitives/tx.js#L418-L472)
+
+### Money Supply
+
+- Base unit: [dollarydoo](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/pkg.js#L52)
+
+- 1 Coin = [1,000,000 dollarydoos](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/protocol/consensus.js#L30)
+
+- Maximum total supply: [2,040,000,000 hns](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/protocol/consensus.js#L111)
+
+- Airdrop supply: [1,360,000,000 hns](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/protocol/consensus.js#L86)
+(Yep, the airdrop is 2/3 of the entire money supply! Available at genesis).
+
+- Initial block subsidy: [2,000 hns](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/protocol/consensus.js#L121)
+
+- Block reward halving interval: [170,000 blocks (about 3 1/4 years)](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/protocol/networks.js#L91)
+
+### Transactions
+
+Because Handshake is SegWit-only, there are no output scripts, just addresses
+(which themselves are witness program version + witness program hash).
+A witness program can encode a public key or a script hash, but those keys and
+scripts will appear in transaction _inputs_.
+
+Handshake also introduces a new field in transaction outputs called a `covenant`.
```

Comment:
Maybe useful to copy this from the white paper:

``` c
struct output_s {
  int64_t value;
  uint8_t script[];
  struct covenant_s {
    uint8_t action;
    uint8_t arguments[][];
  } covenant;
} output;
```

Reply:
The javascript version of that is linked in the next line

Reply:
The covenant type is obscured behind the `covenant.write` method

Reply:
True, but a few lines down I link to the `write` method in covenant.js. What I want to show here is more relevant to transaction serialization, then under covenants section more details are revealed.

Reply:
Ok I'm not super opinionated on this but I do think that seeing the struct is interesting